diff --git a/src/deps/aws-lambda-cpp-1c884fe4392b2d73947d6d6dc8da6a4b146e412b/src/runtime.cpp b/src/deps/aws-lambda-cpp-1c884fe4392b2d73947d6d6dc8da6a4b146e412b/src/runtime.cpp
index a2b14fc..3da8375 100644
--- a/src/deps/aws-lambda-cpp-1c884fe4392b2d73947d6d6dc8da6a4b146e412b/src/runtime.cpp
+++ b/src/deps/aws-lambda-cpp-1c884fe4392b2d73947d6d6dc8da6a4b146e412b/src/runtime.cpp
@@ -179,6 +179,8 @@ void runtime::set_curl_next_options()
     curl_easy_setopt(m_curl_handle, CURLOPT_WRITEFUNCTION, write_data);
     curl_easy_setopt(m_curl_handle, CURLOPT_HEADERFUNCTION, write_header);
 
+    curl_easy_setopt(m_curl_handle, CURLOPT_PROXY, "");
+
 #ifndef NDEBUG
     curl_easy_setopt(m_curl_handle, CURLOPT_VERBOSE, 1);
     curl_easy_setopt(m_curl_handle, CURLOPT_DEBUGFUNCTION, rt_curl_debug_callback);
@@ -199,6 +201,8 @@ void runtime::set_curl_post_result_options()
     curl_easy_setopt(m_curl_handle, CURLOPT_WRITEFUNCTION, write_data);
     curl_easy_setopt(m_curl_handle, CURLOPT_HEADERFUNCTION, write_header);
 
+    curl_easy_setopt(m_curl_handle, CURLOPT_PROXY, "");
+
 #ifndef NDEBUG
     curl_easy_setopt(m_curl_handle, CURLOPT_VERBOSE, 1);
     curl_easy_setopt(m_curl_handle, CURLOPT_DEBUGFUNCTION, rt_curl_debug_callback);
